### PR TITLE
feat: conferencing atoms

### PR DESCRIPTION
**Pull request by ibex088**
## What does this PR do?

- Fixes #XXXX (GitHub issue number)
- Fixes CAL-XXXX (Linear issue number - should be visible at the bottom of the GitHub issue description)

https://www.loom.com/share/fbd40d0bd0d74ef6b8d194d0f4342799?sid=d6eb6d18-91ea-462a-a7ce-5727d3d0194b

## Mandatory Tasks (DO NOT REMOVE)

- [x] I have self-reviewed the code (A decent size PR without self-review might be rejected).
- [x] I have updated the developer docs in /docs if this PR makes changes that would require a [documentation change](https://cal.com/docs). If N/A, write N/A here and check the checkbox.
- [x] I confirm automated tests are in place that prove my fix is effective or that my feature works.

### Prerequisites:
- Add the following redirect URL to your Zoom app in the [[Zoom Marketplace](https://marketplace.zoom.us/)](https://marketplace.zoom.us/):  
  `http://localhost:5555/v2/conferencing/zoom/oauth/callback`

- Need to publish a new vision of platform-libraries

### CODE REVIEW

File: packages/trpc/server/routers/loggedInViewer/integrations.handler.ts
Comment (SomayChauhan) [inline packages/trpc/server/routers/loggedInViewer/integrations.handler.ts:1]: The logic has been relocated to `packages/lib/getConnectedApps.ts` to be shared via `platform-libraries` in `api/v2`.

File: apps/api/v2/src/modules/conferencing/controllers/conferencing.controller.ts [hunk 1]
```diff
@@ -47,25 +71,99 @@ export class ConferencingController {
...
       default:
         throw new BadRequestException(
           "Invalid conferencing app, available apps are: ",
-          CONFERENCING_APPS.join(", ")
+          [GOOGLE_MEET].join(", ")
         );
     }
   }
 
+  @Get("/:app/oauth/auth-url")
```
Comment (SomayChauhan) [inline apps/api/v2/src/modules/conferencing/controllers/conferencing.controller.ts:89]: created a separate endpoint for oAuth conferencing apps

File: apps/api/v2/src/modules/conferencing/controllers/conferencing.controller.ts [hunk 2]
```diff
@@ -47,25 +71,99 @@ export class ConferencingController {
...
+          status: SUCCESS_STATUS,
+          data: plainToInstance(ConferencingAppsOauthUrlOutputDto, credential),
+        };
+
+      default:
+        throw new BadRequestException("Invalid conferencing app, available apps are: ", [ZOOM].join(", "));
+    }
+  }
+
+  @Get("/:app/oauth/callback")
```
Comment (SomayChauhan) [inline apps/api/v2/src/modules/conferencing/controllers/conferencing.controller.ts:124]: oauth conferencing apps callback

File: apps/api/v2/src/modules/conferencing/services/google-meet.service.ts
Comment (SomayChauhan) [inline apps/api/v2/src/modules/conferencing/services/google-meet.service.ts:1]: Both `disconnectGoogleMeetApp` and `setDefault` have been moved to `apps/api/v2/src/modules/conferencing/services/conferencing.service.ts` to handle them consistently across all conferencing apps.

File: apps/api/v2/src/modules/apps/apps.repository.ts [hunk 3]
```diff
@@ -32,4 +41,28 @@ export class AppsRepository {
       },
     });
   }
+
+  async findAppCredintial({
```
Comment (ThyMinimalDev) [inline apps/api/v2/src/modules/apps/apps.repository.ts:45]: typo, Credintial

File: apps/api/v2/src/modules/conferencing/services/conferencing.service.ts [hunk 4]
```diff
@@ -1,14 +1,60 @@
...
+
+  async disconnectConferencingApp(userId: number, app: string) {
+    const credential = await this.checkAppIsValidAndConnected(userId, app);
+    const user = await this.usersRepository.findById(userId);
+    return handleDeleteCredential({ userId, userMetadata: user?.metadata, credentialId: credential.id });
+  }
+
+  async setDefaultConferencingApp(userId: number, app: string) {
+    // cal-video is global, so we can skip this check
+    if (app !== CAL_VIDEO) {
```
Comment (SomayChauhan) [inline apps/api/v2/src/modules/conferencing/services/conferencing.service.ts:50]: cal-video is global, it does not have a credential per-user

File: apps/api/v2/src/modules/conferencing/controllers/conferencing.controller.ts [hunk 5]
```diff
@@ -56,16 +81,89 @@ export class ConferencingController {
...
+  @Redirect(undefined, 301)
+  @ApiOperation({ summary: "conferencing apps oauths callback" })
+  async save(
+    @Query("state") state: string,
+    @Param("app") app: string,
+    @Query("code") code: string,
+    @Query("error") error: string | undefined,
+    @Query("error_description") error_description: string | undefined
+  ): Promise<{ url: string }> {
+    const decodedCallbackState: OAuthCallbackState = JSON.parse(JSON.parse(state));
```
Comment (ThyMinimalDev) [inline apps/api/v2/src/modules/conferencing/controllers/conferencing.controller.ts:135]: JSON parse twice ?

File: apps/api/v2/package.json [hunk 6]
```diff
@@ -31,7 +31,6 @@
     "@calcom/platform-enums": "*",
     "@calcom/platform-libraries": "npm:@calcom/platform-libraries@0.0.69",
```
Comment (ThyMinimalDev) [inline apps/api/v2/package.json:32]: npm:@calcom/platform-libraries@0.0.70

**Review by ThyMinimalDev — APPROVED** [review]
great work

**Review by ThyMinimalDev — APPROVED** [review]
great work

File: apps/api/v2/src/modules/atoms/services/conferencing-atom.service.ts [hunk 7]
```diff
@@ -0,0 +1,25 @@
...
+  constructor(private readonly dbWrite: PrismaWriteService) {}
+
+  async getConferencingApps(user: UserWithProfile): Promise<ConnectedApps> {
+    return getConnectedApps({
+      user,
+      input: {
+        variant: "conferencing",
+        onlyInstalled: true,
+      },
+      prisma: this.dbWrite.prisma as unknown as PrismaClient,
```
Comment (SomayChauhan) [inline apps/api/v2/src/modules/atoms/services/conferencing-atom.service.ts:22]: you mean like this??

**Pull request merged**